plugins/migration/bin/fix-imports.js

```diff
@@ -0,0 +1,246 @@
+  },
+  {
+    import: 'blockly/javascript',
+    oldIdentifier: 'Blockly.JavaScript',
+    newIdentifier: 'javascriptGenerator',
+    newImport: `import {javascriptGenerator} from 'blockly/javascript';`,
+    newRequire: `const {javascriptGenerator} = require('blockly/javascript');`
+  },
+  {
+    import: 'blockly/lua',
+    newIdentifier: 'luaGenerator',
+    newImport: `import {luaGenerator} from 'blockly/lua';`,
+    newRequire: `const {luaGenerator} = require('blockly/lua');`
+  },
+  {
+    import: 'blockly/php',
+    newIdentifier: 'phpGenerator',
+    newImport: `import {phpGenerator} from 'blockly/php';`,
+    newRequire: `const {phpGenerator} = require('blockly/php');`
+  },
+  {
+    import: 'blockly/python',
+    newIdentifier: 'pythonGenerator',
+    newImport: `import {pythonGenerator} from 'blockly/python';`,
+    newRequire: `const {pythonGenerator} = require('blockly/python');`
+  },
+  {
+    import: 'blockly/blocks',
+    oldIdentifier: 'Blockly.libraryBlocks',
+    newIdentifier: 'libraryBlocks',
+    newImport:  `import {libraryBlocks} from 'blockly/blocks';`,
+    newRequire: `const {libraryBlocks} = require('blockly/blocks');`
+  }
+]
+
+/**
+ * Migrates the contents of a particular file, renaming references and
+ * adding/updating imports.
+ * 
+ * @param {string} contents The string contents of the file to migrate.
+ * @return {string} The migrated contents of the file.
+ */
+function migrateContents(contents) {
+  let newContents = contents;
+  for (const migrationData of database) {
+    newContents = fixImport(newContents, migrationData);
+  }
+  return newContents;
+}
+
+/**
+ * Migrates a particular import in a particular file. Renames references to
+ * where the import used to exist on the namespace tree, and adds/updates
+ * imports.
+ * 
+ * @param {string} contents The string contents of the file to migrate.
+ * @param {MigrationData} migrationData Data defining what to migrate and how.
+ * @return {string} The migrated contents of the file.
+ */
+function fixImport(contents, migrationData) {
+  const identifier = getIdentifier(contents, migrationData);
+  if (!identifier) return contents;
+  const newContents = replaceReferences(contents, migrationData, identifier);
+  if (newContents !== contents) return addImport(newContents, migrationData);
+  return contents;
+}
+
+/**
+ * Returns the identifier a given import is assigned to.
+ * 
+ * @param {string} contents The string contents of the file to migrate.
+ * @param {MigrationData} migrationData Data defining what to migrate and how.
+ * @return The identifier associated with the import associated with the
+ *     migration data.
+ */
+function getIdentifier(contents, migrationData) {
+  const importMatch = contents.match(
+      new RegExp(`\\s(\\S*) from ('|")${migrationData.import}('|")`));
+  if (importMatch) return importMatch[1];
+  const requireMatch = contents.match(
+      new RegExp(`(\\S*) = require\\(('|")${migrationData.import}('|")\\)`));
```

Comment:
```suggestion
      new RegExp(`(\\S*)\\s+=\\s+require\\(['"]${migrationData.import}['"]\\)`));
```
(This could yet have a few more `\\s*`s added to it too, if we wanted to be very thorough.)